packages/signals/signals/src/core/signals/signals.ts

```diff
@@ -83,6 +83,12 @@ export class Signals implements ISignals {
    * - Registers custom signal generators.
    */
   async start(analytics: AnyAnalytics): Promise<void> {
+    if (
```

Comment:
Can this live in`AnalyticsService`? i.e. have an `analyticsService.isDisabled()` ? That class is a mean to be a `facade` to abstract away cdnSettings or analytics instance related logic, exposing that to our domain -- for testing, and to keep this initialization stuff tidier.

Reply:
Understanding this feature more, we should have analyticsServie.sampleRate and we should probably have NetworkConfig have a "signalCollectionDisabled": boolean